test/common/upstream/health_check_fuzz.cc

```diff
@@ -0,0 +1,200 @@
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+  reuse_connection_ = true;
+  if (input.health_check_config().has_reuse_connection()) { // TODO: Does this make sense?
+    reuse_connection_ = input.health_check_config().reuse_connection().value();
+  }
+  replay(input);
+}
+
+void HealthCheckFuzz::respondHttp(test::fuzz::Headers headers, absl::string_view status) {
+
+  // Timeout timer needs to be explicitly enabled, usually by onIntervalBase() (Callback on interval
+  // timer).
+  if (!test_sessions_[0]->timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping response.");
+    return;
+  }
+
+  std::unique_ptr<Http::TestResponseHeaderMapImpl> response_headers =
+      std::make_unique<Http::TestResponseHeaderMapImpl>(
+          Fuzz::fromHeaders<Http::TestResponseHeaderMapImpl>(headers, {}, {}));
+
+  response_headers->setStatus(status);
+
+  ENVOY_LOG_MISC(trace, "Responded headers");
+
+  // Responding with http can cause client to close, if so create a new one.
+  bool client_will_close = false;
+  if (response_headers->Connection()) {
+    client_will_close =
+        absl::EqualsIgnoreCase(response_headers->Connection()->value().getStringView(),
+                               Http::Headers::get().ConnectionValues.Close);
+  }
+
+  if (response_headers->ProxyConnection()) {
+    client_will_close =
+        absl::EqualsIgnoreCase(response_headers->ProxyConnection()->value().getStringView(),
+                               Http::Headers::get().ConnectionValues.Close);
+  }
+
+  test_sessions_[0]->stream_response_callbacks_->decodeHeaders(std::move(response_headers), true);
+
+  if (!reuse_connection_ || client_will_close) {
+    ENVOY_LOG_MISC(trace, "Creating client and stream because shouldClose() is true");
+    expectClientCreate(0);
+    expectStreamCreate(0);
+    test_sessions_[0]->interval_timer_->invokeCallback();
+  }
+}
+
+void HealthCheckFuzz::triggerIntervalTimer() {
+  // Interval timer needs to be explicitly enabled, usually by decodeHeaders.
+  if (!test_sessions_[0]->interval_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Interval timer is disabled. Skipping trigger interval timer.");
+    return;
+  }
+  expectStreamCreate(0);
+  ENVOY_LOG_MISC(trace, "Triggered interval timer");
+  test_sessions_[0]->interval_timer_->invokeCallback();
+}
+
+void HealthCheckFuzz::triggerTimeoutTimer(bool last_action) {
+  // Timeout timer needs to be explicitly enabled, usually by a call to onIntervalBase().
+  if (!test_sessions_[0]->timeout_timer_->enabled_) {
+    ENVOY_LOG_MISC(trace, "Timeout timer is disabled. Skipping trigger timeout timer.");
+    return;
+  }
+  ENVOY_LOG_MISC(trace, "Triggered timeout timer");
+  test_sessions_[0]->timeout_timer_->invokeCallback(); // This closes the client, turns off timeout
+                                                       // and enables interval
+  if (!last_action) {
```

Comment:
why does the interval timer need to be invoked with this?

Reply:
Sets up the other events to actually work (with timeout timer enabled, a client and stream to talk with). Wrote about this in my design doc, from one of the comments you left. Otherwise, you'd have to wait for invoke interval timer event to pop up before anything else works.